### PR TITLE
Add generic app hook to consolidate fetch data, clean up queries

### PR DESCRIPTION
**Pull request by justinoboyle**

### CODE REVIEW

File: src/helpers/marketstackHelper.ts [hunk 1]
```diff
@@ -144,3 +146,37 @@ export const persistEODDataForPastNYears = async (
 
   return persistedEodData;
 };
+
+export const doesSecurityExist = async (symbol: string): Promise<boolean> => {
```
Comment (max-zaremba) [inline src/helpers/marketstackHelper.ts:150]: NBD, but mismatch in naming convention between the prisma schema and here, ticker used in prisma nad symbol used here, should use the same

File: src/pages/api/pages/top-level.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+import { constructHandler, InternalResponse } from "../../../helpers/errors";
+
+import type { UserSession } from "../../../helpers/userHelper";
+import { getPortfoliosByUser } from "../../../helpers/portfolioHelper";
+// import { createPortfolio } from "../../../helpers/portfolioHelper";
+
+import type { PortfolioWithPositions } from "../../../helpers/portfolioHelper";
+
+// TODO update
```
Comment (max-zaremba) [inline src/pages/api/pages/top-level.ts:14]: add to todo here maybe? not required

**Review by max-zaremba — COMMENTED** [review]
LGTM for the most part, just check symbol vs ticker discrepency, otherwise you can merge

File: src/pages/api/pages/top-level.ts (hunk 2 above)
Comment (justinoboyle) [reply src/pages/api/pages/top-level.ts:14]: ah i mean when those two features are added, add it to the endpoint

**Pull request merged**